frontend/src/hooks/useHashParam.ts

```diff
@@ -0,0 +1,37 @@
+import { useCallback, useEffect, useState } from "react";
+
+/**
+ * Custom hook to manage a hash parameter in the URL.
+ * @param key - The key of the hash parameter to manage.
+ * @param defaultValue - The default value if the parameter doesn't exist.
+ */
+export function useHashParam(key: string, defaultValue: string) {
```

Comment:
Should we call setValue when defaultValue changes? Also what should happen to the value when key changes?